core-feature-pack/common/src/main/resources/content/bin/systemd/wildfly-domain.service

```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=WildFly (domain mode)
+After=syslog.target network.target
+Before=httpd.service
+
+[Service]
+Environment=LAUNCH_JBOSS_IN_BACKGROUND=1
+Environment="JAVA_HOME=/usr/lib/jvm/jre"
+
+Environment="WILDFLY_SH=/opt/wildfly/bin/domain.sh"
+Environment="WILDFLY_SERVER_CONFIG=domain.xml"
+Environment="WILDFLY_CONSOLE_LOG=/opt/wildfly/domain/log/systemd-console.log"
+Environment="WILDFLY_HOST_CONFIG="
+Environment="WILDFLY_BIND=0.0.0.0"
+EnvironmentFile=-/etc/sysconfig/wildfly-domain.conf
```

Comment:
This is just to make `wildfly-domain.conf` an optional file since the Systemd unit is enough to make the server run. `wildfly-domain.conf` can be optionally used to replace the default values but it is not mandatory. A '-' prefix is not used to validate whether the content of the file is correct, it is just to validate if the file exists.

Notice the documentation link is: https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#EnvironmentFile=

"""
The argument passed should be an absolute filename or wildcard expression, optionally prefixed with "-", which indicates that if the file does not exist, it will not be read and no error or warning message is logged.
"""

I have no strong opinions about this, right now it aligns with the RPM installations done by the product, so the question is more whether we want to make `wildfly-domain.conf` mandatory to start the server even if the system unit is enough to be able to start the server and you have not added any variable declaration.

One possible advantage of making it mandatory is that you will be sure from the beginning that the file you pretend to use for configuration is read. Otherwise, a weird effect is to be modifying it and see how your server starts but you don't see (or understand) why your environment configuration is not being applied. That could occur if you incorrectly place it in an incorrect place.

Reply:
This makes sense to me. I was mostly curious on the intent, but you explained it.